hazelcast/src/main/java/com/hazelcast/internal/adapter/ICacheDataStructureAdapter.java

```diff
@@ -117,8 +117,8 @@ public boolean replace(K key, V oldValue, V newValue) {
     }
 
     @Override
-    public void remove(K key) {
-        cache.remove(key);
+    public V remove(K key) {
+        return cache.getAndRemove(key);
```

Comment:
This is what uncovered the issue (`cache.remove()` works fine, `cache.getAndRemove()` was broken).